### ISSUE TITLE
CP Filters do not work as expected on user field types in collections

### ISSUE THREAD

**Issue by joseph-d** [statamic/cms#3384]
## Bug Description

With a completely fresh install of `statamic/statamic` core, a sample pages collection is created with one sample page. The collection has a blueprint which provides Title, Content, Author, Template, Slug and Parent as fields.

All of the fields can be filtered using the "Filter" select box in the CP except for the "Author" (a `users` field) which does not work as expected.

## How to Reproduce

- Create a completely fresh installation of Statamic with no changes or addons.
- Add a user and log in to the CP
- Go to the "Pages" collection and create some pages, leaving the user selected as the "Author"
- Try to filter by Author, Content, Slug, Template or Title. They all work except for Author
- Selecting "Template" from the filter converts the bottom two filter fields to a single Select containing Template names. However, when Author is selected there is no Select containing Author names as expected.

### Filtering for a template provides a Select containing a list of templates. This is also the case for `collections` or `entries` fields:

![Screenshot_20210316_140747](https://user-images.githubusercontent.com/1070155/111322623-07752700-8661-11eb-8afd-60dd62e6bed3.png)

### Filtering for an Author only provides a text box for filtering which does not work:

![Screenshot_20210316_140712](https://user-images.githubusercontent.com/1070155/111322613-0512cd00-8661-11eb-8e5a-5d7afa66c109.png)

### If you filter by the actual `id` of the Author then it does work:

![Screenshot_20210316_144051](https://user-images.githubusercontent.com/1070155/111327828-a13ed300-8665-11eb-890d-ead677985687.png)

## Thoughts

I mentioned this to support and it was suggested that I might want to submit an idea / feature request if I wanted to be able to filter by author. However, I have chosen to submit this as a bug because actually I don't really require this feature. All I really want is to be able to let end users view the list of entries in a collection with an "Author" (ie: a `user` field) as one of the columns and for all of the entries in the Filters list to work as expected so that end users do not consider it to be broken.

In other words, if there is some reason why a `users` field cannot work in the same was as, say, a `templates`, `collections` or `entries` field then perhaps `users` type fields should be automatically hidden from the Filter menu so that they don't cause confusion to end users who will consider it to be a broken feature. However, given that searching by `id` on a `users` field is currently working, one would assume that filters for users fields are supposed to be rendering a Select field and not a Text field so that the user id can be selected by name.

## Environment

Statamic `3.0.47 Solo`
Laravel `8.32.1`
PHP `8.0.3`
No addons installed

**Install method (choose one):**
Fresh install from `statamic/statamic`

**Comment by robdekort** [comment]
I agree this is more a bug than a FR. However it's as far as I can see a duplicate of #2742. Closing this in favour of that one.
